frontends/mit-open/src/pages/OnboardingPage/TopicInterestsStep.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react"
+import { Grid, Container, ChoiceBox } from "ol-components"
+
+import { useLearningResourceTopics } from "api/hooks/learningResources"
+import Prompt from "./Prompt"
+import { StepProps } from "./types"
+
+function TopicInterestsStep({ onUpdate, profile }: StepProps) {
+  const { data: topics } = useLearningResourceTopics({ is_toplevel: true })
+  const [topicInterests, setTopicInterests] = React.useState<number[]>(
+    profile.topic_interests?.map((topic) => topic.id) || [],
+  )
+
+  const handleToggle = (value: number, checked: boolean) => {
+    setTopicInterests((prevTopicInterests) => {
+      if (checked) {
+        return [...prevTopicInterests, value]
+      } else {
+        return prevTopicInterests.filter((interest) => interest !== value)
+      }
+    })
+  }
+
+  React.useEffect(() => {
+    onUpdate({ topic_interests: topicInterests })
+  }, [topicInterests, onUpdate])
+
+  return profile ? (
+    <>
+      <h3>
+        Welcome, {profile.name}! What are you interested in learning about?
```

Comment:
There was that previous suggestion about using a fallback when profile.name is undefined which I think is worth doing since nothing on the backend guarantees this will be defined. I think the snippet was ```Welcome{profile.name? `, ${profile.name}`: ``}!```